Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -11986,6 +11986,11 @@ void WebPageProxy::requestGeolocationPermissionForFrame(IPC::Connection& connect
     if (!frame)
         return;
 
+#if ENABLE(GEOLOCATION)
+    // Add origin to set of origins that have requested permission to use the Geolocation API.
+    internals().geolocationPermissionRequesters.add(frameInfo.securityOrigin);
```

Comment:
Why is this PR specific to geolocation? Shouldn't we do this for all permission types? Seems like we could replace `geolocationPermissionRequesters` is a more generic HashMap whose key a the permission type (e.g. "geolocation", "notification", ...).